homeassistant/components/wake_on_lan/button.py

```diff
@@ -0,0 +1,87 @@
+"""Support for button entity in wake on lan."""
+
+from __future__ import annotations
+
+from functools import partial
+import logging
+from typing import Any
+
+import wakeonlan
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_BROADCAST_ADDRESS, CONF_BROADCAST_PORT, CONF_MAC
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Wake on LAN sensor entry."""
+    broadcast_address: str | None = entry.options.get(CONF_BROADCAST_ADDRESS)
+    broadcast_port: int | None = entry.options.get(CONF_BROADCAST_PORT)
+    mac_address: str = entry.options[CONF_MAC]
+    name: str = entry.title
+
+    async_add_entities(
+        [
+            WolSwitch(
+                name,
+                mac_address,
+                broadcast_address,
+                broadcast_port,
+            )
+        ]
+    )
+
+
+class WolSwitch(ButtonEntity):
```

Comment:
WolButton

Reply:
Addressed in https://github.com/home-assistant/core/pull/122549